xen/arch/x86/tpm.c

```diff
@@ -340,8 +355,7 @@ static void tpm_hash_extend12(unsigned loc, uint8_t *buf, unsigned size,
         .pcrNum = swap32(pcr),
     };
 
-    if ( out_digest != NULL )
```

Comment:
Why is this removed? `out_digest` could be NULL if we ever run out of space in the event log.

Reply:
Because `sha1_hash(buf, size, (uint32_t *)out_digest);` above would cause a SEGFAULT before this check runs :)  So I concluded the check was left here accidentally.